cradle-core/src/test/java/com/exactpro/cradle/utils/SerializationMessageTest.java

```diff
@@ -0,0 +1,184 @@
+		builder.setContent("Message".repeat(10).getBytes(StandardCharsets.UTF_8));
+		StoredMessage build = builder.build();
+		MessageSerializer serializer = new MessageSerializer();
+
+		ByteBuffer buffer = ByteBuffer.allocate(10_000);
+		serializer.serialize(build, buffer);
+		Assert.assertEquals(buffer.position(), MessagesSizeCalculator.calculateMessageSize(build));
+	}
+
+	static MessageCommonParams getCommonParams() {
+		MessageCommonParams commonParams = new MessageCommonParams();
+		commonParams.setBookId(new BookId("book123456"));
+		commonParams.setSessionAlias("stream_name12345");
+		commonParams.setDirection(Direction.SECOND);
+		return commonParams;
+	}
+
+	static List<StoredMessage> getBatch() {
+		return getBatch(getCommonParams());
+	}
+
+	static List<StoredMessage> getBatch(MessageCommonParams params) {
+
+		StoredMessageBuilder builder = new StoredMessageBuilder();
+		builder.setSessionAlias(params.getSessionAlias());
+		builder.setIndex(123456789010111213L);
+		builder.setDirection(params.getDirection());
+        builder.setBookId(params.getBookId());
+		builder.setTimestamp(Instant.parse("2007-12-03T10:15:30.00Z"));
+		builder.putMetadata("key1", "value1");
+		builder.putMetadata("key2", "value2");
+		builder.putMetadata("key3", "value3");
+		builder.setContent("Message".repeat(10).getBytes(StandardCharsets.UTF_8));
+
+		List<StoredMessage> stMessage = new ArrayList<>(10);
+		stMessage.add(builder.build());
+
+		builder.setIndex(123456789010111214L);
+		builder.setTimestamp(Instant.parse("2007-12-03T10:15:30.01Z"));
+		builder.setContent("Message".repeat(10).getBytes(StandardCharsets.UTF_8));
+		stMessage.add(builder.build());
+
+		builder.setIndex(123456789010111215L);
+		builder.setTimestamp(Instant.parse("2007-12-03T10:15:30.01Z"));
+		builder.setContent("Message".repeat(10).getBytes(StandardCharsets.UTF_8));
+		stMessage.add(builder.build());
+
+		builder.setIndex(123456789010111216L);
+		builder.setTimestamp(Instant.parse("2007-12-03T10:15:30.02Z"));
+		builder.setContent("Message2".repeat(10).getBytes(StandardCharsets.UTF_8));
+		stMessage.add(builder.build());
+
+		builder.setIndex(123456789010111217L);
+		builder.setTimestamp(Instant.parse("2007-12-03T10:15:30.03Z"));
+		builder.setContent("Message3".repeat(10).getBytes(StandardCharsets.UTF_8));
+		stMessage.add(builder.build());
+
+		return stMessage;
+	}
+
+	@Test
+	public void serializeDeserialize5BATCH() throws SerializationException {
+		MessageSerializer serializer = new MessageSerializer();
+		MessageCommonParams commonParams = getCommonParams();
+		List<StoredMessage> initBatch = getBatch(commonParams);
+		byte[] serialize = serializer.serializeBatch(initBatch);
+		MessageDeserializer deserializer = new MessageDeserializer();
+		List<StoredMessage> deserialize = deserializer.deserializeBatch(serialize, commonParams);
+		Assert.assertEquals(deserialize, initBatch);
+	}
+
+	@Test
+	public void serializeDeserialize6EMTPYBATCH() throws SerializationException {
```

Comment:
What means this method name?

Reply:
It means that we are trying to serialize collection with 0 messages and deserialize it.